2_horseshoe_bat/technical_specification.md

```diff
@@ -0,0 +1,199 @@
+# Metadata Submission via REST API (Horseshoe Bat)
+
+## The Scope
+
+### Summary:
+
+The aim of this epic is to extend the "Blobfish" version of the Metadata Repository Service
+to include a submission API.
+
+Moreover, the service should be integrated into the mesh of file-oriented services
+("Red-lipped Batfish" epic).
+
+### Included/Required:
+
+- strategy for data submission interface via spreadsheet
+- implement service that validates a spreadsheet and transforms it into a JSON object
+ready to use for submission to the Metadata Repository Service API
+- extend Metadata Repository Service API to include endpoints used by data submitters
+- publish event upon submission of metadata for new study receive once file has been uploaded
+and is available for download (has a DRS URI)
+
+### Optional:
+
+- metadata update/versioning
+- UI implementation
+
+### Not included:
+
+- user authentication & authorization
+- restricted/encrypted metadata
+
+
+## User Journeys
+
+This epic covers the following user journeys:
+
+
+![](miro_user_journeys.jpeg)
+
+
+### 1.0 Data Submitter submits their metadata spreadsheet to the Metadata Transpiler Service
+
+The Data Submitter submits their metadata spreadsheet to the Metadata
+Transpiler Service (MTS) (1.0).
+
+The MTS first converts the spreadsheet into a Submission JSON.
+
+Then the MTS validates the metadata by making use of the GHGA Metadata Validator (1.1)
+
+- If the validation passes, the prepared Submission JSON is sent back to the
+Data Submitter (1.2)
+- If the validation fails, then the Data Submitter is notified (1.2) along 
+with the validation errors
+
+
+### 2.0 Data Submitter submits a Study as part of a Submission
+
+The Data Submitter submits the Submission object to the Metadata Repository Service (MRS)
+as a HTTP POST request.
+
+The MRS validates the incoming Submission object, and all of the associated entities.
+
+Upon successful validation, the MRS assigns a Submission ID (unique identifier) to the Submission.
+Then the Submission object, and its associated entities, are written to the metadata store (2.1).
+
+Upon a successful write operation, the Submission object (which includes a Submission ID) is
+returned to the Data Submitter for reference. If the validation fails then the data submitter
+is notified of a failure (2.2).
+
+**Note:** A Submission can have one of the following status: `in progress`, `submitted`, 
+`released`. At this stage, the Submission is neither considered 'submitted' nor
```

Comment:
Maybe we don't even need a released status because only the data set not the submission is released, right?

Reply:
Agreed.